parser.py

```diff
@@ -1,17 +1,30 @@
 #!/bin/env python
 import argparse
 import pprint
+import re
+
 import grammar
 import visitor
 
+class PreProcessor:
```

Comment:
Absolute nitpick but `Preprocessor` is a single word. This camel-casing makes me a bit itchy. :confused: